services/workflows-service/src/workflow/workflow.controller.external.ts

```diff
@@ -318,6 +318,8 @@ export class WorkflowControllerExternal {
       currentProjectId,
     );
 
+    const hasSalesforceRecord = 'salesforceObjectName' in body && 'salesforceRecordId' in body;
```

Comment:
What if their value is `undefined` or an empty string?

Reply:
Empty string is ok, but `undefined` is not. It's a client input, how can it be set to `undefined`?

Reply:
Let me re-phrase. Is validation not needed there beyond the existence of property?

Reply:
We didn't enable `exactOptionalPropertyTypes` so I changed the condition a bit. Thanks.